server/src/main/java/io/deephaven/server/arrow/ArrowFlightUtil.java

```diff
@@ -513,351 +540,53 @@ private void tryClose() {
-                final RowSet viewport =
-                        hasViewport ? BarrageProtoUtil.toRowSet(subscriptionRequest.viewportAsByteBuffer()) : null;
-
-                final boolean reverseViewport = subscriptionRequest.reverseViewport();
-
-                final boolean subscriptionFound;
-                if (bmp != null) {
-                    subscriptionFound = bmp.updateSubscription(listener, viewport, columns, reverseViewport);
-                } else if (htvs != null) {
-                    subscriptionFound = true;
-                    htvs.setViewport(columns, viewport, reverseViewport);
-                } else {
-                    subscriptionFound = false;
-                }
+    /**
+     * Helper to retrieve the viewport RowSet from a subscription request.
+     */
+    public static @Nullable RowSet getViewport(final BarrageSubscriptionRequest subscriptionRequest) {
+        final boolean hasViewport = subscriptionRequest.viewportVector() != null;
+        return hasViewport ? BarrageProtoUtil.toRowSet(subscriptionRequest.viewportAsByteBuffer()) : null;
+    }
 
-                if (!subscriptionFound) {
-                    throw Exceptions.statusRuntimeException(Code.NOT_FOUND, "Subscription was not found.");
-                }
-            }
+    /**
+     * Helper to retrieve the viewport columns from a subscription request.
+     */
+    public static @Nullable BitSet getColumns(final BarrageSubscriptionRequest subscriptionRequest) {
+        final boolean hasColumns = subscriptionRequest.columnsVector() != null;
+        return hasColumns ? BitSet.valueOf(subscriptionRequest.columnsAsByteBuffer()) : null;
+    }
 
-            @Override
-            public synchronized void close() {
-                if (bmp != null) {
-                    bmp.removeSubscription(listener);
-                    bmp = null;
-                } else if (htvs != null) {
-                    htvs.completed();
-                    htvs = null;
-                } else {
-                    GrpcUtil.safelyComplete(listener);
-                }
+    /**
+     * Helper to retrieve the viewport RowSet from a snapshot request.
+     */
+    static @Nullable RowSet getViewport(final BarrageSnapshotRequest snapshotRequest) {
+        final boolean hasViewport = snapshotRequest.viewportVector() != null;
+        return hasViewport
+                ? BarrageProtoUtil.toRowSet(snapshotRequest.viewportAsByteBuffer())
+                : null;
+    }
 
-                // After we've signaled that the stream should close, cancel the export
-                if (onExportResolvedContinuation != null) {
-                    onExportResolvedContinuation.cancel();
-                    onExportResolvedContinuation = null;
-                }
+    /**
+     * Helper to retrieve the viewport columns from a snapshot request.
+     */
+    static @Nullable BitSet getColumns(final BarrageSnapshotRequest snapshotRequest) {
```

Comment:
Should these two static helpers also be public? (getViewport + getColumns on the BarrageSnapshotRequest?)

Reply:
Yes, and I think it makes sense to put them in their own top-level class too.